configure.ac

```diff
@@ -2562,20 +2526,17 @@ yes)
     Darwin*)
         # -Wno-long-double, -no-cpp-precomp, and -mno-fused-madd
         # used to be here, but non-Apple gcc doesn't accept them.
-        if test "${CC}" = gcc
-        then
-            AC_MSG_CHECKING([which compiler should be used])
-            case "${UNIVERSALSDK}" in
-            */MacOSX10.4u.sdk)
-                # Build using 10.4 SDK, force usage of gcc when the
-                # compiler is gcc, otherwise the user will get very
-                # confusing error messages when building on OSX 10.6
-                CC=gcc-4.0
-                CPP=cpp-4.0
-                ;;
-            esac
-            AC_MSG_RESULT([$CC])
-        fi
+        AC_MSG_CHECKING([which compiler should be used])
+        case "${UNIVERSALSDK}" in
+        */MacOSX10.4u.sdk)
+            # Build using 10.4 SDK, force usage of gcc when the
+            # compiler is gcc, otherwise the user will get very
+            # confusing error messages when building on OSX 10.6
+            CC=gcc-4.0
+            CPP=cpp-4.0
+            ;;
+        esac
+        AC_MSG_RESULT([$CC])
```

Comment:
Consider reverting this check and instead base it on $CC_BASENAME.

Reply:
Since our current minimum is apparently [macOS 10.9](https://www.python.org/downloads/release/python-3123/), maybe this entire block can be removed.

Reply:
Believe it or not, current Pythons are still being built for and used on very old macOS systems (see, for example, the MacPorts project) but, to do so, one needs to use newer compilers than those supplied with the original releases (like gcc-4.0). So I agree, we should just remove this block.